botocore/compress.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import gzip
+import io
+import logging
+
+from botocore.utils import determine_content_length
+
+logger = logging.getLogger(__name__)
+
+
+class RequestCompressor:
```

Comment:
What's our reasoning for the class here given we have no constants or shared state? It looks like we're only exposing a single method (`compress`).

Reply:
This is discussed in the design. Its for organization and extensibility purposes. While it may not happen any time soon, additional compression encodings could be modeled in the future and require the SDK's support.

Reply:
Per offline discussion I've updated to a functional implementation.